front/src/common/Map/Layers/InfraObjectLayers/InfraObjectLayers.tsx

```diff
@@ -18,84 +19,101 @@ type InfraObjectLayersProps = {
   infraId: number;
   mapStyle: MapStyle;
   hoveredOperationalPointId?: string;
+  layersSettings: LayersSettings;
 };
 
 const InfraObjectLayers = ({
   infraId,
   mapStyle,
   hoveredOperationalPointId,
+  layersSettings,
 }: InfraObjectLayersProps) => (
   <>
     <TracksGeographic
       colors={colors[mapStyle]}
       layerOrder={LAYER_GROUPS_ORDER[LAYERS.TRACKS.GROUP]}
       infraID={infraId}
     />
-
     <Routes
```

Comment:
Routes also need to be displayed only if its setting is toggled (see in the Routes component ;) )